src/components/Rating/index.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+
+const Rating = ({ initialRating, maxRating, ratingEmoji }) => {
+  const [rating, setRating] = useState(initialRating);
+  const [hover, setHover] = useState(null);
+  const emojiElements = [...Array(maxRating || 5)];
+
+  let emoji;
+  switch (ratingEmoji) {
+    case 'heart': {
+      emoji = '❤️';
+
+      break;
+    }
+    case 'smiley': {
+      emoji = '🙂';
+
+      break;
+    }
+    case 'star': {
+      emoji = '⭐';
+
+      break;
+    }
+    case 'money': {
+      emoji = '💰';
+
+      break;
+    }
+    default: {
+      emoji = '👍🏻';
+    }
+  }
+
+  return (
+    <div className="rating">
+      { emojiElements.map((_key, index) => {
+        const ratingValue = index + 1;
+
+        return (
+          <label key={ index }>
+            <input
+              type="radio"
+              className="rating__radio"
+              value={ ratingValue }
+              onClick={ () => setRating(ratingValue) }
+            />
+            <span
+              role="img"
+              aria-label="emoji"
+              className={
+                ratingValue <= (hover || rating)
+                  ? 'rating__emoji rating__emoji--selected'
+                  : 'rating__emoji'
+              }
+              onMouseEnter={ () => setHover(ratingValue) }
```

Comment:
@rafayet-monon How about we use style to make the hovered item got `opacity:1`? 🤔  just like the example from [here](https://stackoverflow.com/questions/21196588/css-hover-effect-on-current-and-previous-elements/21196863) so we don't need to use the hover state